AutoRest/Generators/CSharp/Azure.CSharp.Tests/Expected/AcceptanceTests/AzureParameterGrouping/ParameterGroupingOperations.cs

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator 0.11.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+namespace Fixtures.Azure.AcceptanceTestsAzureParameterGrouping
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest;
+    using Newtonsoft.Json;
+    using Microsoft.Rest.Azure;
+    using Models;
+
+    /// <summary>
+    /// ParameterGroupingOperations operations.
+    /// </summary>
+    internal partial class ParameterGroupingOperations : IServiceOperations<AutoRestParameterGroupingTestService>, IParameterGroupingOperations
+    {
+        /// <summary>
+        /// Initializes a new instance of the ParameterGroupingOperations class.
+        /// </summary>
+        /// <param name='client'>
+        /// Reference to the service client.
+        /// </param>
+        internal ParameterGroupingOperations(AutoRestParameterGroupingTestService client)
+        {
+            if (client == null) 
+            {
+                throw new ArgumentNullException("client");
+            }
+            this.Client = client;
+        }
+
+        /// <summary>
+        /// Gets a reference to the AutoRestParameterGroupingTestService
+        /// </summary>
+        public AutoRestParameterGroupingTestService Client { get; private set; }
+
+        /// <summary>
+        /// Post a bunch of required parameters grouped
+        /// </summary>
+        /// <param name='parameterGroupingPostRequiredParameters'>
+        /// Additional parameters for the operation/// </param>
+        /// <param name='customHeaders'>
+        /// Headers that will be added to request.
+        /// </param>
+        /// <param name='cancellationToken'>
+        /// The cancellation token.
+        /// </param>
+        public async Task<AzureOperationResponse> PostRequiredWithHttpMessagesAsync(ParameterGroupingPostRequiredParameters parameterGroupingPostRequiredParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var parameterGroupingPostRequiredParametersBody = (parameterGroupingPostRequiredParameters == null ? default(int?) : parameterGroupingPostRequiredParameters.Body);
```

Comment:
The name of the variable should probably be the same as in swagger (unless it's not unique)